lib/msf/core/db_manager/exploit_attempt.rb

```diff
@@ -27,186 +27,145 @@ def report_exploit_attempt(host, opts)
   }
   end
 
+  # Create an `Mdm::ExploitAttempt` (and possibly an `Mdm::VulnAttempt`, if
+  # the `vuln` option is passed).
+  #
+  # @option (see #do_report_failure_or_success)
+  # @return (see #do_report_failure_or_success)
   def report_exploit_failure(opts)
+    return unless opts.has_key?(:refs) && !opts[:refs].blank?
```

Comment:
Thanks for changing this from `opts.delete`.  It wasn't nice deleting the caller's `Hash`.